### PR TITLE
Feat#1287: Improving the search index API & API helper modifications in Feedback service

### PR DESCRIPTION
**Pull request by ghanlohar**
# Closes/Fixes/Resolves
ONEPLAT-1287
### 

# Explain the feature/fix

- Improving the search index API 
- Search API helper modifications in Feedback service

## Does this PR introduce a breaking change
Yes

## Screenshots

<details>
<summary>View Screenshots</summary>

Screenshot 2021-07-10 at 2 14 28 AM

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/search-service/src/search-config/typedef.graphql [hunk 1]
```diff
@@ -69,10 +69,14 @@ type Query {
 }
 
 type Mutation {
+  """
+  Manage index creation updation & deletion
+  """
+  manageIndex(input: SearchInput, mode: String): SearchResponseCode
```
Comment (riginoommen) [inline packages/search-service/src/search-config/typedef.graphql:75]:
```suggestion
  manageIndex(input: SearchInput, mode: String!): SearchResponseCode
```

File: packages/search-service/src/search-mapping/resolver.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ export const SearchMapResolver = {
         triggerSearchMap ( root: any, args: any, ctx: any ) {
             const searchMapCron = new SearchMapCron();
             searchMapCron.searchMapTrigger();
+            return "Indexing Search Maps has been triggered.";
```
Comment (riginoommen) [inline packages/search-service/src/search-mapping/resolver.ts:15]:
```suggestion
            return 'Indexing Search Maps has been triggered.';
```

File: packages/feedback-service/src/helpers.ts [hunk 3]
```diff
@@ -337,42 +337,35 @@ class FeedbackHelper {
...
             method: `POST`,
             headers,
-            body: body,
-        }).then((response: any) => response.json())
-            .then((result: any) => {
-                if ((result.data?.manageIndex?.status === 200) || (result?.data?.deleteIndex?.status === 204)) {
+            body: JSON.stringify({
+                query: query,
+                variables: {
+                    input: mode === "index"? data?.input : data,
```
Comment (riginoommen) [inline packages/feedback-service/src/helpers.ts:357]:
```suggestion
                    input: mode === 'index'? data?.input : data,
```

File: packages/feedback-service/src/helpers.ts [hunk 4]
```diff
@@ -337,42 +337,35 @@ class FeedbackHelper {
...
+        } ).then( ( response: any ) => {
+            return response.json();
+        } )
+            .then( ( result: any ) => {
+                const statuses = [ 200, 204 ];
+                if ( statuses.includes(result.data?.manageIndex?.status) ) {
                     console.log('Sucessfully completed the index updation')
-                } else if ((result?.data?.manageIndex?.status !== 200) || (result?.data?.removeIndex?.status !== 204)) {
+                } else if ( !statuses.includes(result.data?.manageIndex?.status) ) {
                     console.log("Error in index updation.");
```
Comment (riginoommen) [inline packages/feedback-service/src/helpers.ts:369]:
```suggestion
                    console.info('Error in index updation.');
```

File: packages/feedback-service/src/helpers.ts [hunk 5]
```diff
@@ -337,42 +337,35 @@ class FeedbackHelper {
...
+                    mode
+                }
+            } ),
+        } ).then( ( response: any ) => {
+            return response.json();
+        } )
+            .then( ( result: any ) => {
+                const statuses = [ 200, 204 ];
+                if ( statuses.includes(result.data?.manageIndex?.status) ) {
                     console.log('Sucessfully completed the index updation')
```
Comment (riginoommen) [inline packages/feedback-service/src/helpers.ts:367]:
```suggestion
                    console.info('Sucessfully completed the index updation')
```

File: packages/feedback-service/src/helpers.ts [hunk 6]
```diff
@@ -325,7 +325,7 @@ class FeedbackHelper {
                     'description': data.description,
                     'icon': `assets/icons/feedback.svg`,
                     'uri': `${process.env.FEEDBACK_CLIENT}`,
-                    'tags': `Feedback App, ${data?.category},`,
+                    'tags': `Feedback, ${data?.category},`,
```
Comment (riginoommen) [inline packages/feedback-service/src/helpers.ts:328]:
```suggestion
                    'tags': `Feedback, ${data?.category}`,
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/feedback-service/src/helpers.ts (hunk 3 above)
Comment (ghanlohar) [reply packages/feedback-service/src/helpers.ts]: Done

File: packages/search-service/src/search-config/typedef.graphql (hunk 1 above)
Comment (ghanlohar) [reply packages/search-service/src/search-config/typedef.graphql]: Done

File: packages/search-service/src/search-mapping/resolver.ts (hunk 2 above)
Comment (ghanlohar) [reply packages/search-service/src/search-mapping/resolver.ts]: Done

File: packages/feedback-service/src/helpers.ts (hunk 4 above)
Comment (ghanlohar) [reply packages/feedback-service/src/helpers.ts]: Done

File: packages/feedback-service/src/helpers.ts (hunk 5 above)
Comment (ghanlohar) [reply packages/feedback-service/src/helpers.ts]: Done

File: packages/feedback-service/src/helpers.ts (hunk 6 above)
Comment (ghanlohar) [reply packages/feedback-service/src/helpers.ts]: Done

**Review by riginoommen — APPROVED** [review]

**Review by deshmukhmayur — APPROVED** [review]

**Pull request merged**